sql-plugin/src/main/scala/com/nvidia/spark/rapids/NoopWriteSupport.scala

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import scala.reflect.ClassTag
+import scala.util.Try
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.connector.catalog.SupportsWrite
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.datasources.v2.LeafV2CommandExec
+import org.apache.spark.sql.vectorized.ColumnarBatch
+import org.apache.spark.util.Utils
+
+/**
+ * Support for noop data source writes that creates non-V1 write executors.
+ * These executors might not exist in all Spark versions, so we use conditional loading.
+ */
+object NoopWriteSupport extends Logging {
+  
+  // Class names for the non-V1 write executors
+  private val overwriteByExpressionExecClassName = 
+    "org.apache.spark.sql.execution.datasources.v2.OverwriteByExpressionExec"
+  private val appendDataExecClassName = 
+    "org.apache.spark.sql.execution.datasources.v2.AppendDataExec"
+    
+  // Check if the non-V1 write executors are available
+  lazy val hasNonV1WriteExecs: Boolean = {
+    Utils.classIsLoadable(overwriteByExpressionExecClassName) && 
```

Comment:
You are getting compile errors here because `org.apache.spark.util.Utils` is private to apache spark. In addition to this I don't want to use reflection to pull these things apart we want to reference the classes directly when ever possible. Like we do with the rest of the code.